benchopt/runner.py

```diff
@@ -205,6 +205,77 @@ def get_next(x): return max(x / rho, MIN_TOL)
     return curve, status
 
 
+class Callback:
+
+    def __init__(self, objective, max_iter, deadline, meta):
+        """Get callback function and store informations for the curve.
+
+        Parameters
+        ----------
+        objective : instance of BaseObjective
+            The objective to minimize.
+        max_iter : int
+            Maximum number of iterations to run for.
+        deadline : float
+            Deadline after which to stop the computations. This will be
+            used to respect the timeout for each solver.
+        meta : dict
+            Metadata passed to store in Cost results.
+            Contains objective and data names, problem dimension, etc.
+        """
+
+        self.objective = objective
+        self.max_iter = max_iter
+        self.deadline = deadline
+        self.meta = meta
+        self.info = get_sys_info()
+        self.curve = []
+        self.cb_status = {
+                'status': 'unfinished',
+                "it": 0,
+                'next_stopval': 0,
```

Comment:
why do you need to store next_stopval? I would just keep it as attribute of Callback.